src/query-builder/UpdateQueryBuilder.ts

```diff
@@ -642,11 +642,16 @@ export class UpdateQueryBuilder<Entity extends ObjectLiteral>
                 if (
                     metadata.updateDateColumn &&
                     updatedColumns.indexOf(metadata.updateDateColumn) === -1
-                )
-                    updateColumnAndValues.push(
-                        this.escape(metadata.updateDateColumn.databaseName) +
-                            " = CURRENT_TIMESTAMP",
-                    ) // todo: fix issue with CURRENT_TIMESTAMP(6) being used, can "DEFAULT" be used?!
+                ) {
+                    const updateDateColumn = metadata.updateDateColumn;
+                    let defaultValue = updateDateColumn.default;
+                    if (typeof defaultValue === "function") {
+                        defaultValue = defaultValue();
+                    }
+                    updateColumnAndValues.push(this.escape(updateDateColumn.databaseName) +
+                        " = " +
+                        defaultValue);
```

Comment:
I would somehow do this:
```ts
valuesSetNormalized[updateDateColumn.databaseName] = updateDateColumn.default
```
and let the `else` block handle the whole thing 🤷🏻‍♂️

Why? `defaultValue` needs to be handled properly, e.g. if it's not a function it needs to be escape. Currently this is already implemented in the `else` block.